### DISCUSSION TITLE
GTK window-decoraction=false corners

### DISCUSSION THREAD

**Discussion by rhodes-b** [ghostty-org/ghostty#3290] [Feature Requests, Ideas] [answered]
It seems that we still se rounded edges with gtk when setting window-decoration=false

There are many users that have this I will post a few including my own. I am using cinnamon so I am fine with the default behavior as it is the gnome look (brucespruceog in the discord just trying to help some others)

**Comment by rhodes-b** [comment]
![image](https://github.com/user-attachments/assets/d13135a9-3cf0-4e8a-a8da-cf1b826d7b90)

**Comment by rhodes-b** [comment]
![image](https://github.com/user-attachments/assets/f11374cb-316f-44a8-8864-810aea78e7fd)

**Reply by rhodes-b** [reply]
zoom in on this one

**Comment by rhodes-b** [comment]
![image](https://github.com/user-attachments/assets/34f37880-647e-4e74-a0b6-3676632bb998)

**Comment by rhodes-b** [comment]
tagging @jcollie @tristan957 for awareness I think the css might have something weird or this is just how gtk behaves im not sure of another app that supports the non rounded edges (like I said im fine with it)

**Reply by tristan957** [reply]
Are you using libadwaita

**Reply by rhodes-b** [reply]
Yes I am, I would assume the others are as well, for now ill keep recommending that to other people though to disable adwaita

**Reply by rhodes-b** [reply]
Just tried with `gtk-adwaita = false` still seems like it is there, although it is pretty small (and only the top corners?)
makes a significant difference for the bottom corners perhaps this should be the default is window-decoration = false?

**Comment by rhodes-b** [comment]
ill close this, seems like DE can get involved at least with cinnamon potentially others as well

**Reply by rhodes-b** [reply]
Tristan and I have some more info here
https://discord.com/channels/1005603569187160125/1322024110359838731

**Reply by ndavd** [reply]
@rhodes-b Can you share some details here? I wasn't able to open that and I'm also facing this issue.

**Reply by rhodes-b** [reply]
@ndavd 
seems like DE can get involved at least with cinnamon potentially others as well

**Reply by ndavd** [reply]
What does that mean?

**Reply by ndavd** [reply]
All my other applications respond well to the WM. I've never had any issues like this.

**Reply by rhodes-b** [reply]
Its your window manager or Desktop environment applying styling to the windows (other tiling wm do not have this issue so yours is applying some styling to it somehow)

**Reply by ndavd** [reply]
None of my other applications have rounded corners, only this one. My WM has no rounded corners, just straight ones with the red border as I show in the image bellow.
![image](https://github.com/user-attachments/assets/9ac338c0-2714-4e87-b6a8-aad80b6b72e3)

If I show Kitty terminal, for example, you see it doesn't create that artifact in the corner:
![image](https://github.com/user-attachments/assets/7f6416ea-0629-4f0d-a65e-52b11f457527)

**Reply by rhodes-b** [reply]
https://github.com/ghostty-org/ghostty/blob/6508fec945258f526e0adf0dcc174d954e23fb8d/src/apprt/gtk/Window.zig#L211

the gtk css should not be in play here with adwaita disabled, as we explicitly tell gtk not to style the window

**Reply by ndavd** [reply]
Not sure how I can help, but it's definitely an issue with ghostty.
Other terminals, like Kitty, show no artifacts:
![image](https://github.com/user-attachments/assets/f8d31e2d-9f6f-409b-a89c-ae621f0186a8)

**Reply by rhodes-b** [reply]
Other terminals, like Kitty, show no artifacts
other terminals are not using gtk

**Reply by ndavd** [reply]
Not sure what else to say if you think it's not an issue with ghostty. I'll keep using kitty, not sure what else I can tweak on my system, literally no application in my system has rounded corners apart from this one.

**Reply by rhodes-b** [reply]
are you using any other gtk apps on your system

**Reply by tristan957** [reply]
Set `gtk-adwaita=false`. Unless you are using adwaita, this is not a ghostty issue. Kitty is a good terminal, and fine to keep using it.

**Reply by tristan957** [reply]
Actually just put the following CSS in your global GTK CSS file:

```css
window {
  border-radius: 0 0;
}
```

**Reply by tristan957** [reply]
![image](https://github.com/user-attachments/assets/b27a164c-3144-4139-abef-fe228d6c76a4)

**Reply by ndavd** [reply]
That CSS worked indeed @tristan957, thanks! I'm using Adwaita, yeah. So just that CSS adjustment fixed my issue.

**Comment by tristan957** [comment, marked as answer]
Howdy all. Please see https://github.com/ghostty-org/ghostty/pull/3999 for what is probably a proper fix for everyone, so in theory you could remove your custom CSS now :)

**Reply by ndavd** [reply]
I removed the custom CSS and it works, thanks! 🙏🏻

**Comment by ncfavier** [comment]
I am still seeing rounded corners with Ghostty 1.1.0, with all of

```
window-decoration = false
gtk-titlebar = false
gtk-adwaita = false
```

EDIT: probably related: https://github.com/ghostty-org/ghostty/issues/2023
